src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -22,11 +22,59 @@
 class YamlFileDumper extends FileDumper
 {
     /**
+     * Dump messages as tree, e.g. this
+     *   'foo.bar1' => 'test1',
+     *   'foo.bar2' => 'test2'
+     *
+     * dumps as follows:
+     *   foo:
+     *     bar1: test1
+     *     bar2: test2
+     *
+     * @var bool
+     */
+    private $asTree = true;
+
+    /**
+     * The level where you switch to inline YAML.
+     * Do not switch to inline YAML by default.
+     *
+     * @var int
+     */
+    private $inline = 999;
+
+    /**
+     * Sets asTree flag
+     * 
+     * @param bool $asTree
+     */
+    public function setDumpAsTree($asTree)
```

Comment:
I'd prefer `2.` it's not a BC break but it must be documented in the UPGRADE file.